sympy/stats/crv_types.py

```diff
@@ -364,11 +370,28 @@ def pdf(self, x):
         alpha, beta = self.alpha, self.beta
         return x**(alpha - 1) * (1 - x)**(beta - 1) / beta_fn(alpha, beta)
 
-    def sample(self, size=()):
-        if not size:
-            return random.betavariate(self.alpha, self.beta)
-        else:
-            return [random.betavariate(self.alpha, self.beta)]*size
+    def _sample_python(self, size):
+        x = Symbol('x')
+        a, b = float(self.alpha), float(self.beta)
+        if isinstance(size, int):
+            return ArrayComprehensionMap(lambda: random.betavariate(a, b), (x, 0, size-1)).doit()
+        return ArrayComprehensionMap(lambda: random.betavariate(a, b),
+                                     *[(x, 0, i-1) for i in size]).doit()
+
+    def _sample_numpy(self, size):
+        a, b = float(self.alpha), float(self.beta)
+        return numpy.random.beta(a=a, b=b, size=size)
+
+    def _sample_scipy(self, size):
+        a, b = float(self.alpha), float(self.beta)
+        from scipy.stats import beta
+        return beta.rvs(a=a, b=b, size=size)
+
+    def _sample_pymc3(self, size):
+        a, b = float(self.alpha), float(self.beta)
+        with pymc3.Model():
+            pymc3.Beta('X', alpha=a, beta=b)
+            return pymc3.sample(size, chains=1, progressbar=False)[:]['X']
```

Comment:
Similarly more methods will have to be added for new libraries. Can we generalise this too?

Reply:
I think we can generalize this as this becomes easy to continue according to current API design in the future. But still, I will look into it in more detail if I could come up with some other better way.